### ISSUE TITLE
Outlook Web - Send event for Smart Alerts stops runtime for other events.

### ISSUE THREAD

**Issue by ArkaiAriza** [OfficeDev/office-js#3530]
## Your Environment
* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: N/A
* Operating System: Windows 10
* Browser (if using Office on the web): Chrome v114
 
## Expected behavior
After triggering Smart Alerts and selecting Don't Send, the handlers for other events like OnMessageCompose should keep running.

## Current behavior
Currently, the handlers for other events stop when calling event.completed() for the Send event Smart Alerts.

## Steps to reproduce
1. Have an add in with an OnSend event handler for Smart Alerts and another event handler like OnMessageCompose.
2. For testing purposes, make it so the Smart Alert triggers when sending a message.
3. Make it so the OnMessageCompose handler runs some code you can monitor. A console.log in an interval is enough.
4. Compose a new message. The console logs from OnMessageCompose handler should appear.
5. Try to send the email. Smart Alerts popup should appear.
6. Select Don't Send in the Smart Alerts popup.
7. Console logs from OnMessageCompose handler will stop.

## Link to live example(s)
1. If needed I should be able to provide a minimal example code.

## Context
Basically I want to monitors the contents of the message, and alert under certain conditions. For this I use OnMessageCompose event and OnSend event with Smart Alerts. I'm pretty sure this wasn't the behaviour a couple of weeks ago. I can't provide a better timespan as I noticed Monday, July 17. In Outlook for Windows it works as expected.

## Useful logs
No noticeable errors.

**Comment by exextoc** [comment]
Hi @ArkaiAriza, thanks for reporting this. Could you please confirm if our understanding of your use-case is correct here:
- We open a new mail compose form and OnMessageCompose triggers. It runs a piece of code and console logs in certain intervals.
- We then try to send the email when the OnSend addins blocks the send and the code that was triggered by OnMessageCompose also stops executing at that time.

Also, it'd help us greatly if you could share a sample add-in to replicate the same. You may email us at: OutlookExtensibility@M365x27035562.onmicrosoft.com

**Comment by ArkaiAriza** [comment]
Hi @exextoc . Yes, that is exactly our use case. Just want to emphasize that it wasn't like this a couple of weeks ago and it works well in Outlook for Windows. I'll try to get a sample add-in and send it to that email address. Thanks you very much for your attention.

**Comment by exextoc** [comment]
Hi @ArkaiAriza, thanks for sharing the add-in. The launch-event based add-ins do have some limitations. Kindly refer to the documentation here: https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/autolaunch?tabs=xmlmanifest#event-based-activation-behavior-and-limitations

We would like to highlight a couple of points from this documentation:
1. After activation, your add-in will time out within approximately 300 seconds, the maximum length of time allowed for running event-based add-ins.
2. The code included after the event.completed statement isn't guaranteed to run

Relating this to your use-case, once the event-handler for OnNewMessageCompose is triggered, it is set to timeout within 300 seconds. Also, since event.completed is called has part of the onMessageSendHandler - it can't be guaranteed that the event handler launched at the time of OnNewMessageCompose continues to run.

**Comment by ArkaiAriza** [comment]
Hi @exextoc thanks for your feedback. I am aware of the limitations you mention and I have considered them before:

1. The issue I'm reporting is not produced by a timeout, as It can be reproduced without hiting the 300 seconds limit. 
2. 
- The documentation does not suggest that calling event.completed for an event handler would cancel the execution of ongoing event handlers for other different events (the docs refer to code included after the call). If this were to be the case, I think it would at least need an specific warning in the documentation, because people handling different events with different handlers would need to be very careful so their handlers do not inadvertently cancel each other out.
- In the same note, my use case was working well for several months until a couple of weeks ago.
- Finally, there is at least a consistency issue, as my use case is currently working in Outlook for Windows.

Thank you very much again for your attention, and I hope my answers and the code sample I sent were helpful.

**Comment by exextoc** [comment]
OWA had a bug because of which this behavior was being supported, but we have fixed it and it cannot be supported now because of limitations. Please refer to this document which clearly states that launch event handlers should be short-running and lightweight. [Configure your Outlook add-in for event-based activation - Office Add-ins | Microsoft Learn](https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/autolaunch?tabs=xmlmanifest#event-based-activation-behavior-and-limitations). Also, you can let us know the exact user case for which timer is being used so that a work-around can be suggested.
